modules/objdetect/src/face_detect.cpp

```diff
@@ -102,12 +102,21 @@ class FaceDetectorYNImpl : public FaceDetectorYN
             return 0;
         }
         CV_CheckEQ(input_image.size(), Size(inputW, inputH), "Size does not match. Call setInputSize(size) if input size does not match the preset size");
-        // Pad input_image with divisor 32
-        Mat pad_image = padWithDivisor(input_image);
-
-        // Build blob from input image
-        Mat input_blob = dnn::blobFromImage(pad_image);
 
+        Mat input_blob;
+        if(input_image.kind() == _InputArray::UMAT) {
+            // Pad input_image with divisor 32
+            UMat pad_image;
+            padWithDivisor(input_image, pad_image);
+            // Build blob from input image
+            input_blob = dnn::blobFromImage(pad_image);
+        } else {
```

Comment:
@vpisarev T-API declares what we should not have such code separation on "user" side.